deps/NNNN-container-strategy.md

```diff
@@ -0,0 +1,338 @@
+# Proposal
+
+In order to address the requirements, we propose the following changes to the Dynamo build process:
+
+## Build-Base Container
+
+The base container will be a pre-built container that will be used by the backends to build the final container image. This build base container will contain a Dynamo build for all backends to use for their framework-specific build. The base image will leverage a manylinux base image to enable support for multiple distributions (U22, U24, etc). The container will also include a NIXL build since this is common across all backends. This will be a new Dockerfile in the /containers directory. Multi-arch support is also a P0 Also, the base container can be used as a drop-in replacement for the current devel container used in public CI. This would significantly reduce public CI build times and enable faster validation of changes. 
+
+## Use-case of build stages along with relationship between stages (base, runtime, devel, ci_minimum)
+
+Each backend-specific Dockerfile should follow a specific format. The backend-specific Dockerfiles should be divided up into multiple stages, with each stage inheriting artifacts/leveraging the previous stage as the base container. The following stages should be defined in the backend-specific Dockerfile: 
+
+| Stage    | Targeted User                | Base Image           | Functionality                                                                                                         |
+|----------|---------------------|----------------------|----------------------------------------------------------------------------------------------------------------------|
+| Backend Build    | Developers          | Cuda base devel image     | Builds targeted backend along with backend-specific dependencies in editable mode. 
+| Runtime  | Customers/Production| Cuda base runtime image| Minimal image with only the dependencies required to deploy and run Dynamo w/backend from the backend build stage; intended for production deployments. Copies dynamo artifacts from base image and backend artifaces from backend build image. |
+| CI       | Developers/Internal CI Pipelines/Local Debugging | Runtime image          | Adds CI-specific tools, QA test scripts, internal models, and other dependencies needed for automated testing.         |
+
+The CUDA base images will be used from the [NVIDIA CUDA Container Registry](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/cuda). Please refer to the Pros and Cons section for more details on why we chose to use the cuda runtime image instead of the Deep Learning CUDA image.
+
+# Implementation Details
+
+## Container Build Flow
+
+```mermaid
+flowchart TD
+    A[Manylinux build base image]:::gray
+    B[NIXL Setup/Build NIXL Wheel]
+    C[Build Dependencies]
+    D[Build Dynamo]
+    E[Dynamo Base Container]:::gray
+    F[Build Backend from source]
+    G[Backend Build Image]:::gray
+    J[Cuda Runtime<br/>nvcr.io/nvidia/cuda.XX.YY-runtime]:::gray
```

Comment:
selecting a color is always risky due to user's local experience.

example:

<img width="625" height="371" alt="Image" src="https://github.com/user-attachments/assets/665bf8d7-69c1-40fe-82bd-5efe351e2748" />

Reply:
Changing this or not is your call. I just wanted to call this out and provide the information.